### DISCUSSION TITLE
With a multi tenancy panel, how do you allow some pages/resources to have tenant-free routes?

### DISCUSSION THREAD

**Discussion by kurucu** [filamentphp/filament#7992] [General]
I have a panel which is set up for multi-tenancy, to allow most resources to belong to a Team.

But I want a page (or a resource) to sit outside the tenancy system, and be accessible even if the user is not a member of any Team (yet). In this case, it's because it's so the user can see the invitations they have to various teams, and accept them! Using the routing as-is, Filament injects a default tenant, even if none is provided, to the page. If the user has no teams, then the panel breaks, and the page can never be visited.

I have set `isTenantSubscriptionRequired` to return false. Is there anything else I should do?

The way I've implemented this at the moment is a Page in the panel has a single widget, which gets invitations and allows them to be accepted/deleted. I am open to changing this to a resource if it's easier.

**Comment by pboivin** [comment]
It sounds like the custom page is working for you. Another option is to add your resource in a new panel without tenancy. You can easily create links between panels to help your users navigate without having to think about tenancy.

**Reply by kurucu** [reply]
I was hoping it could all be in the same panel, but maybe that's a good idea - different panels.

**Reply by pboivin** [reply]
Some people have had success modifying the resource and page classes to remove the tenant considerations but it's not really straightforward... I would call it "fighting the framework" at this point. I imagine it could be improved in the future.

**Reply by danharrin** [reply]
Proposed in #7525 but the author closed for some reason

**Reply by kurucu** [reply]
Reading that, it looks like they closed it as they couldn't see how to proceed, not because they didn't want it.

**Reply by kurucu** [reply]
So this was the idea I went for in the end - it was quick and easy to do, and much more logical really. So there's a personal panel for invites and an overview of member organisations, and an organisation panel that only has tenant-aware resources and pages. Both have notifications enabled, and are accessible from each other by the user menu. That said, I expect that'll be one way - the default will be to end up in the personal panel and click through, via a widget on that dashboard, to the tenant desired.

**Comment by pslxx** [comment]
update v3.029 All the menu all have no, can't find any resources

**Reply by danharrin** [reply]
Update to .30

**Reply by devhoussam1998** [reply]
I want to restrict some routes from multi tenancy?

**Comment by brain82** [comment]
For pages, you can probably try something like this in your Panel Provider...

`$panel->routes(fn () => Route::get('create-pin', CreatePin::class)->middleware(['signed'])->name('auth.create-pin'))`
